MapboxCoreNavigation/NavigationService.swift

```diff
@@ -12,43 +12,123 @@ public enum SimulationOption: Int {
     case onPoorGPS, always, never
 }
 
-
+/**
+ A `NavigationService` is the entry-point protocol for MapboxCoreNavigation. It contains all the dependancies needed by the `MapboxNavigation` UI SDK, as well as dependancies for it's child objects. If you would like to implement your own core-navigation stack, be sure to conform to this protocol.
+ 
+ */
 @objc(MBNavigationService)
 public protocol NavigationService: CLLocationManagerDelegate, RouterDataSource, EventsManagerDataSource {
+    /**
+     The services' location manager. This will be the object responsible for notifying the service of GPS updates.
+     */
     var locationManager: NavigationLocationManager { get }
+    
+    /**
+     A reference to a MapboxDirections service. Used for rerouting.
+     */
     var directions: Directions { get }
+    
+    /**
+     The active router, responsible for all route-following.
+     */
     var router: Router! { get }
+    
+    /**
+     The EventsManager, responsible for all telemetry.
```

Comment:
Nit: “the events manager”. The technical class name is already part of the declaration. This is where we translate the class name into plain English.